Lib/asyncio/tasks.py

```diff
@@ -635,11 +634,17 @@ def ensure_future(coro_or_future, *, loop=None):
             raise ValueError('The future belongs to a different loop than '
                             'the one specified as the loop argument')
         return coro_or_future
-    called_wrap_awaitable = False
+    should_close = False
     if not coroutines.iscoroutine(coro_or_future):
         if inspect.isawaitable(coro_or_future):
+            async def _wrap_awaitable(awaitable):
+                @types.coroutine
+                def wrapper():
+                    return (yield from awaitable.__await__())
+                return await wrapper()
```

Comment:
```suggestion
                return await awaitable
```

or is the aim to support:

`obj.__await__ = lambda: asyncio.sleep(0, result='ok').__await__()`?